OpenSim/Common/SmoothSegmentedFunction.cpp

```diff
@@ -163,55 +498,92 @@ _x0(x0),_x1(x1),_y0(y0),_y1(y1),_dydx0(dydx0),_dydx1(dydx1),
- {
-        _arraySplineUX.resize(0);        
-        _mXVec.resize(0);
-        _mYVec.resize(0);
-        _splineYintX = SimTK::Spline();
-        _numBezierSections = (int)SimTK::NaN;
-       
- }
-
- /*Detailed Computational Costs
- ________________________________________________________________________
-    If x is in the Bezier Curve
-                            Name     Comp.   Div.    Mult.   Add.    Assign.
+SmoothSegmentedFunction::SmoothSegmentedFunction(
+    const SimTK::Matrix& mX,
+    const SimTK::Matrix& mY,
+    double x0,
+    double x1,
+    double y0,
+    double y1,
+    double dydx0,
+    double dydx1,
+    bool computeIntegral,
+    bool intx0x1,
+    const std::string& name):
+    _smoothData(
+        SmoothSegmentedFunctionDataLookup(
+            SmoothSegmentedFunctionParameters(
+                mX,
+                mY,
+                x0,
+                x1,
+                y0,
+                y1,
+                dydx0,
+                dydx1,
+                computeIntegral,
+                intx0x1),
+            name)
+        ),
+    _name(name)
+{}
+
+SmoothSegmentedFunction::SmoothSegmentedFunction():
+    _smoothData(
+        SmoothSegmentedFunctionDataLookup(
+            SmoothSegmentedFunctionParameters(),
+            "NOT_YET_SET")
+        ),
+    _name("NOT_YET_SET")
+{}
+
+/*Detailed Computational Costs
+________________________________________________________________________
+   If x is in the Bezier Curve
+                           Name     Comp.   Div.    Mult.   Add.    Assign.
 _______________________________________________________________________
-        SegmentedQuinticBezierToolkit::
-                        calcIndex     3*m+2                   1*m     3   
-                            *calcU     15      2      82      42      60  
-        calcQuinticBezierCurveVal                     21      20      13
-                            total  15+3*m+2    2      103     62+1*m  76
+       SegmentedQuinticBezierToolkit::
+                       calcIndex     3*m+2                   1*m     3
+                           *calcU     15      2      82      42      60
+       calcQuinticBezierCurveVal                     21      20      13
+                           total  15+3*m+2    2      103     62+1*m  76
 
-        *Approximate. Uses iteration
+       *Approximate. Uses iteration
 ________________________________________________________________________
 If x is in the linear region
 
-                        Name     Comp.   Div.    Mult.   Add.    Assign.
-                                    1               1      2     1
+                       Name     Comp.   Div.    Mult.   Add.    Assign.
+                                   1               1      2     1
 ________________________________________________________________________
- 
- */
+
+*/
 
 double SmoothSegmentedFunction::calcValue(double x) const
 {
+    auto& arraySplineUX = _smoothData->_arraySplineUX;
```

Comment:
When using references or pointers, always keep `const-` qualification in mind. This should be:

```c++
const auto& arraySplineUX = _smoothData->arraySplineUX;
```

And the typename isn't very long, so you can make it more readable in isolation by removing `auto`:

```c++
const SimTK::Array_<SimTK::Spline>& arraySplineUX = _smoothData->_arraySplineUX;
```